### PR TITLE
tests: fix osd-scrub-repair.sh

### PR DESCRIPTION
**Pull request by dzafman**

### CODE REVIEW

**Comment by dzafman** [comment]
There be only 1 more bug before this pull request can be merged:

http://tracker.ceph.com/issues/17966

There is only the 1 failure on the last Jenkins run.

**Comment by dzafman** [comment]
Ready to be reviewed and merged.

**Comment by dachary** [comment]
@dzafman I propose an alternate fix for  0b1d42a at https://github.com/ceph/ceph/pull/12095

**Comment by dachary** [comment]
When running the test from within an emacs buffer, 
<pre>
[loic@rex001 build]$ stty -a | head -1 | sed -e 's/.*columns \([0-9]*\).*/\1/'                                                                                                 
0                                                                                                                                                                              
[loic@rex001 build]$ stty -a                                                                                                                                                   
speed 38400 baud; rows 0; columns 0; line = 0;                                                                                                                                 
intr = ^C; quit = ^\; erase = <undef>; kill = <undef>; eof = ^D; eol = <undef>; eol2 = <undef>; swtch = <undef>; start = ^Q; stop = ^S; susp = ^Z; rprnt = ^R; werase = ^W;    
lnext = ^V; flush = ^O; min = 1; time = 0;                                                                                                                                     
-parenb -parodd cs8 -hupcl -cstopb cread -clocal -crtscts                                                                                                                      
-ignbrk -brkint -ignpar -parmrk -inpck -istrip -inlcr -igncr icrnl ixon -ixoff -iuclc -ixany -imaxbel -iutf8                                                                   
opost -olcuc -ocrnl -onlcr -onocr -onlret -ofill -ofdel nl0 cr0 tab0 bs0 vt0 ff0                                                                                               
isig icanon iexten -echo echoe echok -echonl -noflsh -xcase -tostop -echoprt echoctl echoke                                                                                    
[loic@rex001 build]$ 
</pre>
and the test fail with 
<pre>
/home/loic/ceph-ubuntu-14.04-loic/src/test/osd/osd-scrub-repair.sh:1811: TEST_corrupt_scrub_erasure:  diff -y -W 0 td/osd-scrub-repair/checkcsjson td/osd-scrub-repair/csjson
diff: invalid width '0'
diff: Try 'diff --help' for more information.
/home/loic/ceph-ubuntu-14.04-loic/src/test/osd/osd-scrub-repair.sh:1811: TEST_corrupt_scrub_erasure:  test no = yes
/home/loic/ceph-ubuntu-14.04-loic/src/test/osd/osd-scrub-repair.sh:1811: TEST_corrupt_scrub_erasure:  return 1
</pre>

**Comment by dachary** [comment]
Now running 
<pre>
while : ; do echo -n . ; ctest -R osd-scrub-repair.sh >& ../out || break ; sleep 1 ; done
</pre>
and I'll let it run until tomorrow morning.

File: src/osd/ECBackend.cc [hunk 1]
```diff
@@ -2348,14 +2348,14 @@ void ECBackend::be_deep_scrub(
...
+      if (hinfo->get_total_chunk_size() != pos) {
+	dout(0) << "_scan_list  " << poid << " got incorrect size on read" << dendl;
+	o.ec_size_mismatch = true;
 	return;
       }
 
-      if (hinfo->get_total_chunk_size() != pos) {
-	dout(0) << "_scan_list  " << poid << " got incorrect size on read" << dendl;
+      if (hinfo->get_chunk_hash(get_parent()->whoami_shard().shard) != h.digest()) {
+	dout(0) << "_scan_list  " << poid << " got incorrect hash on read" << dendl;
```
Comment (tchaikov) [inline src/osd/ECBackend.cc]: this commit looks good to me.

**Pull request merged**